internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -21,54 +33,242 @@ var (
+
+	// Setup bidrectional streaming session.
+	callOpts := []grpc.CallOption{}
+	cstream, err := c.SetUpSession(ctx, callOpts...)
+	if err != nil  {
+		log.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err)
+	}
+	log.Printf("Client: set up bidirectional streaming RPC session.")
+
+	// Send request to S2Av2 for config.
+	err = cstream.Send(&s2av2pb.SessionReq {
+		AuthenticationMechanisms: []*s2av2pb.AuthenticationMechanism {
+			{
+				MechanismOneof: &s2av2pb.AuthenticationMechanism_Token{"token"},
+			},
+		},
+		ReqOneof: &s2av2pb.SessionReq_GetTlsConfigurationReq {
+			&s2av2pb.GetTlsConfigurationReq {
+				ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_CLIENT,
+			},
+		},
+	})
 	if err != nil {
-		log.Fatalf("Failed to generate X509KeyPair: %v", err)
+		log.Fatalf("Client: failed to send SessionReq: %v", err)
+	}
+	log.Printf("Client: sent SessionReq")
+
+	// Get the response containing config from S2Av2.
+	resp, err := cstream.Recv()
+	if err != nil {
+		log.Fatalf("Client: failed to recieve SessionResp: %v", err)
+	}
+	log.Printf("Client: recieved SessionResp")
+
+	// TODO(rmehta19): Handle resp.GetStatus().
+
+	// Extract TLS configiguration from SessionResp.
+	tlsConfig := resp.GetGetTlsConfigurationResp().GetClientTlsConfiguration()
+
+	// Fake S2Av2 only puts one cert in CertitificateChain, for loop iterates
+	// once.
+	var certList []tls.Certificate
+	for _, v := range tlsConfig.CertificateChain {
+		// TODO(rmehta19): Call remote signer library for private key.
+		cert, err := tls.X509KeyPair([]byte(v), clientKey)
+		if err != nil {
+			log.Fatalf("Failed to generate X509KeyPair: %v", err)
+		}
+		certList = append(certList, cert)
 	}
 
 	rootCertPool := x509.NewCertPool()
 	rootCertPool.AppendCertsFromPEM(serverCert)
 
-	// TODO(rmehta19): Call S2Av2 for config values.
+	// Map S2Av2 TLSVersion to consts defined in tls package.
+	var minVersion uint16
+	var maxVersion uint16
+	switch x := tlsConfig.MinTlsVersion; x {
+	case commonpb.TLSVersion_TLS_VERSION_UNSPECIFIED:
+		minVersion = tls.VersionTLS13 // use tls 1.3 in this case.
+	case commonpb.TLSVersion_TLS_VERSION_1_0:
+		minVersion = tls.VersionTLS10
+	case commonpb.TLSVersion_TLS_VERSION_1_1:
+		minVersion = tls.VersionTLS11
+	case commonpb.TLSVersion_TLS_VERSION_1_2:
+		minVersion = tls.VersionTLS12
+	case commonpb.TLSVersion_TLS_VERSION_1_3:
+		minVersion = tls.VersionTLS13
+	default:
+		minVersion = tls.VersionTLS13
+	}
+
+	switch x := tlsConfig.MaxTlsVersion; x {
+	case commonpb.TLSVersion_TLS_VERSION_UNSPECIFIED:
+		maxVersion = tls.VersionTLS13 // use tls 1.3 in this case.
+	case commonpb.TLSVersion_TLS_VERSION_1_0:
+		maxVersion = tls.VersionTLS10
+	case commonpb.TLSVersion_TLS_VERSION_1_1:
+		maxVersion = tls.VersionTLS11
+	case commonpb.TLSVersion_TLS_VERSION_1_2:
+		maxVersion = tls.VersionTLS12
+	case commonpb.TLSVersion_TLS_VERSION_1_3:
+		maxVersion = tls.VersionTLS13
+	default:
+		maxVersion = tls.VersionTLS13
+	}
```

Comment:
Should we do a sanity check to make sure minVersion <= maxVersion?

Reply:
Will do, thanks!